proposal/package-importer.d.ts.md

```diff
@@ -0,0 +1,502 @@
+addition, a file may have access to multiple `node_modules` directories, and
+different files may have access to different `node_modules` directories in the
+same compilation.
+
+There are a variety of methods currently in use for specifying a location of the
+default Sass export for npm packages. For the most part, packages contain both
+JavaScript and styles, and use the `main` or `module` root keys to define the
+JavaScript entry point. Some packages use the `"sass"` key at the root of their
+`package.json`.
+
+Other packages have adopted [conditional exports], driven by build tools like
+[Vite], [Parcel] and [Sass Loader for Webpack] which all resolve Sass paths
+using the `"sass"` and the `"style"` custom conditions.
+
+[conditional exports]: https://nodejs.org/api/packages.html#conditional-exports
+[Vite]: https://github.com/vitejs/vite/pull/7817
+[Parcel]: https://github.com/parcel-bundler/parcel/blob/2d2400ded4615375ee6bd53ef77b4857ad1591dd/packages/transformers/sass/src/SassTransformer.js#L163
+[Sass Loader for Webpack]: https://github.com/webpack-contrib/sass-loader/blob/02df41203adfda96959e56abb43bd35a89ec11ba/src/utils.js#L514
+
+Because use of conditional exports is flexible and recommended for modern
+packages, this will be the primary method used for the Node package importer. We
+will support both the `"sass"` and the `"style"` conditions, as Sass can also
+use the CSS exports exposed through `"style"`. While in practice, `"style"`
+tends to be used solely for `css` files, we will support `scss`, `sass` and
+`css` files for either `"sass"` or `"style"`.
+
+While conditional exports allows package authors to define specific aliases to internal
+files, we will still use the Sass conventions for resolving file paths with
+partials, extensions and indices to discover the intended export alias. However,
+we will not apply that logic to the destination, and will expect library authors
+to map the export to the correct place. In other words, given a `package.json`
+with `exports` as below, The Node package importer will resolve a
+`@use "pkg:pkgName/variables";` to the destination of the `_variables.scss` export.
+
+```json
+{
+  "exports": {
+    "_variables.scss": {
+      "sass": "./src/sass/_variables.scss"
+    }
+  }
+}
+```
+
+## Types
+
+```ts
+import '../spec/js-api';
+```
+
+### `pkgImporter`
+
+If set, the compiler will use the specified built-in package importer to resolve
+any URL with the `pkg:` scheme. Currently, the only available package importer
+is `node`, which follows Node resolution logic to locate Sass files.
+
+Defaults to undefined.
+
+After the first bullet points in [`compile`] and [`compileString`] in the
+Javascript Compile API, insert:
+
+* If `options.pkgImporter` equals `'node'`:
+
+  * Let `pkgImporter` be a [Node Package Importer] with an associated
+    `entryPointURL` of `require.main.filename`.
+
+  * Append `pkgImporter` to the `options.importers`.
```

Comment:
@nex3 Here I'm mutating `options.importers` directly which feels incorrect from a programming standpoint, but sufficiently clear from a specification standpoint. Creating an intermediary combined list is possible, but would be a much more intrusive patch, but let me know if that would be preferable.

Reply:
I added a requirement to add it to a copy of `options.importers`.